ssl/ssl_transfer_asn1.cc

```diff
@@ -485,11 +494,17 @@ static int SSL3_STATE_from_bytes(SSL *ssl, CBS *cbs, const SSL_CTX *ctx) {
       !CBS_get_asn1_uint64(&s3, &early_data_reason) ||
       early_data_reason > ssl_early_data_reason_max_value ||
       !CBS_get_asn1(&s3, &previous_client_finished, CBS_ASN1_OCTETSTRING) ||
-      CBS_len(&previous_client_finished) != PREV_FINISHED_MAX_SIZE ||
+      (is_pre_v3 &&
+       CBS_len(&previous_client_finished) != PRE_V3_PREV_FINISHED_MAX_SIZE) ||
+      (!is_pre_v3 &&
+       CBS_len(&previous_client_finished) != PREV_FINISHED_MAX_SIZE) ||
```

Comment:
Not sure if i'm understanding this correctly, but should `previous_client_finished` in the v3 case always be equal to `PREV_FINISHED_MAX_SIZE`? Should the check not be > MAX_SIZE? Depending on hash function used, the output length could be shorter than PREV_FINISHED_MAX_SIZE (which looks like is 64 to account for SHA-512)?